clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -33,10 +37,37 @@
     private Map<String, String> headers = new HashMap<>();
     private ConnectionState connectionState = null;
     private HttpClient httpClient;
+    private Timer pingTimer = null;
+    private long nextServerTimeout = 0;
```

Comment:
> Reads and writes are atomic for reference variables and for most primitive variables (all types **except long** and double). - https://docs.oracle.com/javase/tutorial/essential/concurrency/atomic.html

You'll need to use [`AtomicLong`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLong.html)